Ginger/GingerCoreNET/Run/RunSetActions/RunSetActionHTMLReportSendEmail.cs

```diff
@@ -156,327 +156,344 @@ public string MailUser
-            else
-            {
-                if (loggerMode == ExecutionLoggerConfiguration.DataRepositoryMethod.TextFile)
+                catch (Exception ex)
                 {
-                    if (ReportItem != null && !WorkSpace.Instance.RunsetExecutor.RunSetConfig.RunsetExecLoggerPopulated)
-                    {
-                        Errors = "In order to get HTML report, please, perform executions before";
-                        Reporter.HideStatusMessage();
-                        Status = Ginger.Run.RunSetActions.RunSetActionBase.eRunSetActionStatus.Failed;
-                        return;
-                    }
+                    Reporter.ToLog(eLogLevel.ERROR, "Failed to send mail", ex);
+                    isSuccess = false;
                 }
-                mValueExpression.Value = Bodytext;
-                if (ReportItem != null)
+                if (isSuccess == false)
                 {
-                    emailReadyHtml = "Full Report Shared Path =>" + reportsResultFolder + System.Environment.NewLine;
+                    Errors = Email.Event;
+                    Reporter.HideStatusMessage();
+                    Status = eRunSetActionStatus.Failed;
                 }
-                emailReadyHtml += mValueExpression.ValueCalculated;
             }
-            //Reporter.ToLog(eLogLevel.DEBUG, "Run set operation send Email: Preparing email");
-            mValueExpression.Value = MailFrom;
-            Email.MailFrom = mValueExpression.ValueCalculated;
-            mValueExpression.Value = MailTo;
-            Email.MailTo = mValueExpression.ValueCalculated;
-            mValueExpression.Value = MailCC;
-            Email.MailCC = mValueExpression.ValueCalculated;
-            mValueExpression.Value = Subject;
-            Email.Subject = mValueExpression.ValueCalculated;
-            mValueExpression.Value = MailHost;
-            Email.SMTPMailHost = mValueExpression.ValueCalculated;
-            mValueExpression.Value = MailUser;
-            Email.SMTPUser = mValueExpression.ValueCalculated;
-            Email.Body = emailReadyHtml;
-            emailReadyHtml = string.Empty;
-            bool isSuccess = false;
-            try
-            {
-                //Reporter.ToLog(eLogLevel.DEBUG, "Run set operation send Email: Before send email");
-                isSuccess = Email.Send();
-                //Reporter.ToLog(eLogLevel.DEBUG, "Run set operation send Email: After send email result = " + isSuccess);
-            }
-            catch (Exception ex)
+            finally
             {
-                Reporter.ToLog(eLogLevel.ERROR, "Failed to send mail", ex);
-                isSuccess = false;
-            }
-            if (isSuccess == false)
-            {
-                Errors = Email.Event;
-                Reporter.HideStatusMessage();
-                Status = eRunSetActionStatus.Failed;
+                if (string.IsNullOrEmpty(tempFolder) == false && Directory.Exists(tempFolder))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unnecessary Boolean literal(s).](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5264586)